core/src/main/java/hudson/model/Computer.java

```diff
@@ -1132,7 +1132,7 @@ public void interrupt() {
 
     @Override
     public String getSearchUrl() {
-        return getUrl();
+        return Jenkins.get().getRootUrl() + getUrl();
```

Comment:
This is not a good fix. Consider Jenkins being used on another than the configured canonical URL. You'll be taken to a login screen, if it works at all (depending on network setup). And I would not consider `gerRootUrlFromRequest()` an appropriate fix either: this problem appears to be introduced in this PR and shouldn't need a long time API to change behavior incompatibly. Consider https://github.com/search?q=org%3Ajenkinsci%20getSearchUrl&type=code

Reply:
Thanks - updated this to not change the API. Results are now prefixed with the root url via JS.